src/Makefile

```diff
@@ -4,10 +4,14 @@ unexport LANGUAGE
 MAKEFLAGS += --no-builtin-rules
 .SUFFIXES:
 
+# Setup common folder structure variables
+mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+SDIR = $(dir $(mkfile_path))
+export
+
 ###########################################
 ## Provide some convenient make targets  ##
 ###########################################
-ifndef  TARGET
 TARGET   ?= devo8
 default: $(TARGET)
 
```

Comment:
We should get rid of the default.  running make without a target should just fail.